crates/stages/src/stages/headers.rs

```diff
@@ -119,27 +119,25 @@ where
         // reverse from there. Else, it should use whatever the forkchoice state reports.
         let target = match next_header {
             Some(header) if checkpoint + 1 != header.number => SyncTarget::Gap(header),
-            None => self.next_sync_target(head_num).await,
+            None => self
+                .next_sync_target(head_num)
+                .await
+                .ok_or(StageError::StageCheckpoint(checkpoint))?,
             _ => return Err(StageError::StageCheckpoint(checkpoint)),
         };
 
         Ok(SyncGap { local_head, target })
     }
 
-    async fn next_sync_target(&mut self, head: BlockNumber) -> SyncTarget {
+    async fn next_sync_target(&mut self, head: BlockNumber) -> Option<SyncTarget> {
```

Comment:
I originally wanted to return a result but the signature doesn't provide enough information for creating a `StageError::StageCheckpoint`, so the Option is looked at to return an error in the event we return `None`.  This is possible because a channel closure was previously unhandled, now it will return `None` and then return a `StageError`.

Reply:
> This is possible because a channel closure was previously unhandled, now it will return None and then return a StageError.

okay I see, because we never handled the error case here (sender is dropped)